src/utils/order.ts

```diff
@@ -272,6 +272,32 @@ export const mapOrderAmountsFromUnitsToFill = (
   };
 };
 
+export function adjustTipsForPartialFills(
+  tips: ConsiderationItem[],
+  unitsToFill: BigNumberish,
+  totalSize: BigNumber,
+): ConsiderationItem[] {
+  const unitsToFillBn = BigNumber.from(unitsToFill);
+
+  if (unitsToFillBn.lte(0)) {
+    throw new Error("Units to fill must be greater than 0");
+  }
+
+  return tips.map((tip) => ({
+    ...tip,
+    startAmount: multiplyDivision(
+      tip.startAmount,
+      unitsToFillBn,
+      totalSize,
+    ).toString(),
```

Comment:
I think the seaport considers tip as extra money you want to pass to someone at a time of fulfillment, irrespective of a number of fulfillment. so it don't become to specific.

Cosider scenario

1. 2 piece listed
2. person tries to fulfill 1 piece with 1 ETH tip
3. With this code, person will receive 0.5 as a tip. (curator fee)

As per convention, whatever you pass needs to be spent. here we are passing 1 eth as a tip but only 0.5 eth is being spent. so why don't we pass 0.5 eth tip in the first place and doing all that calculation on frontend when passing tip.  at a time of fulfillment.